pkg/test/local/policybackend.go

```diff
@@ -0,0 +1,154 @@
+//  Copyright 2018 Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package local
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/davecgh/go-spew/spew"
+	"github.com/gogo/protobuf/proto"
+	"github.com/golang/protobuf/jsonpb"
+
+	"istio.io/istio/pkg/test/environment"
+	"istio.io/istio/pkg/test/fakes/pb"
+	"istio.io/istio/pkg/test/internal"
+)
+
+type policyBackend struct {
+	port       int
+	backend    *pb.PolicyBackend
+	controller *pb.Controller
+}
+
+var _ environment.DeployedPolicyBackend = &policyBackend{}
+
+func newPolicyBackend(ctx *internal.TestContext) (*policyBackend, error) {
+	port := pb.DefaultControlPort
```

Comment:
Do we want to make the port dynamic to avoid collisions?

Reply:
I don't think it matters much (unless we start to instances side-by-side when running locally). Still, I've plumbed it through from the top.